pkg/operator/ceph/csi/spec.go

```diff
@@ -313,3 +322,53 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 	}
 	return nil
 }
+
+// ValidateCSIVersion checks if the configured ceph-csi image is supported
+func ValidateCSIVersion(clientset kubernetes.Interface, namespace, csiPluginImage, rookImage, serviceAccountName string) error {
+	// skip check if running unsupported images is allowed
+	if CSIParam.AllowUnsupported {
+		logger.Infof("Allowing unsupported ceph-csi images -> no version check")
+		return nil
+	}
+
+	ownerRef := metav1.OwnerReference{
+		APIVersion: fmt.Sprintf("%s", "detect-csi-version"),
+		Kind:       "VersionDetection",
+		Name:       namespace,
+		UID:        types.UID(namespace),
+	}
+	timeout := 15 * time.Minute
+
+	logger.Infof("detecting the ceph csi image version for image %s", csiPluginImage)
+	versionReporter, err := cmdreporter.New(
+		clientset, &ownerRef,
+		detectCSIVersionName, detectCSIVersionName, namespace,
+		[]string{"cephcsi"}, []string{""},
+		rookImage, csiPluginImage)
+
+	if err != nil {
+		return fmt.Errorf("failed to set up ceph CSI version job. %+v", err)
+	}
+
+	job := versionReporter.Job()
+	job.Spec.Template.Spec.ServiceAccountName = serviceAccountName
+
+	// CephCSI has not --version or --help parameter
+	// To get the version of it is necessary to start with no or invalid parameters and parse the stderr
```

Comment:
If ceph-csi is adding a version flag, sounds best to wait for it

Reply:
https://github.com/ceph/ceph-csi/pull/704 added version flag in cephcsi

Reply:
I will leave and adopt this PR till this fix will get released.